File: sample-commons/src/main/java/net/xpece/commons/sample/MainActivity.java
package net.xpece.commons.sample;

import android.content.Context;
import android.content.res.ColorStateList;
import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.DatePicker;
import android.widget.TimePicker;

import net.xpece.android.content.XpContext;
import net.xpece.android.content.res.Dimen;
import net.xpece.android.widget.XpDatePicker;
import net.xpece.android.widget.XpEdgeEffect;
import net.xpece.android.widget.XpTimePicker;
import net.xpece.commons.android.sample.R;

import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;

/**
 * Created by Eugen on 25. 4. 2015.
 */
public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ViewPager pager = (ViewPager) findViewById(R.id.pager);
        PagerAdapter adapter = new MyPagerAdapter();
        pager.setAdapter(adapter);

        XpEdgeEffect.setColor(pager, Color.RED);

        Dimen d = Dimen.dp(16);
        Log.d(TAG, "Dimension real size: " + d.toString(this));

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);

        Drawable icon = XpContext.resolveDrawable(toolbar.getContext(), R.attr.homeAsUpIndicator);
        toolbar.setNavigationIcon(icon);
        String title = XpContext.resolveString(this, R.style.Widget_AppCompat_ActionButton_Overflow, android.R.attr.contentDescription);
        toolbar.setTitle(title);
    }

    static class MyPagerAdapter extends PagerAdapter {

        @Override
        public int getCount() {
            return 4;
        }

        @Override
        public boolean isViewFromObject(View view, Object object) {
            return view == object;
        }

        @Override
        public Object instantiateItem(ViewGroup container, int position) {
            final Context context = container.getContext();
            final View view;
            if (position == 0) {
                final ColorStateList csl = ColorStateList.valueOf(Color.RED);
                final TimePicker tp = new TimePicker(context);
                XpTimePicker.setSelectionDividerTint(tp, csl);
                view = tp;
            } else if (position == 1) {
                final ColorStateList csl = ColorStateList.valueOf(Color.RED);
                final DatePicker dp = new DatePicker(context);
                XpDatePicker.setSelectionDividerTint(dp, csl);
                view = dp;
            } else {
                view = new View(context);
            }
            container.addView(view);
            return view;
        }

        @Override
        public void destroyItem(ViewGroup container, int position, Object object) {
            final View view = (View) object;
            container.removeView(view);
        }
    }

    @Override
    protected void attachBaseContext(final Context newBase) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
    }
}

File: sample-commons/src/main/java/net/xpece/commons/sample/MainActivity.kt
package net.xpece.commons.sample

import android.content.Intent
import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.Typeface
import android.graphics.Typeface.BOLD
import android.os.Build.VERSION.SDK_INT
import android.os.Bundle
import android.text.style.BulletSpan
import android.text.style.StrikethroughSpan
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.DatePicker
import android.widget.TextView
import android.widget.TimePicker
import androidx.appcompat.app.AppCompatActivity
import androidx.core.text.buildSpannedString
import androidx.core.text.inSpans
import androidx.viewpager.widget.PagerAdapter
import com.google.android.material.snackbar.Snackbar
import net.xpece.android.app.SnackbarActivity
import net.xpece.android.content.dp
import net.xpece.android.edgeeffect.widget.setEdgeEffectColorCompat
import net.xpece.android.graphics.drawable.RippleDrawableCompatInflater
import net.xpece.android.picker.widget.setSelectionDividerTintCompat
import net.xpece.android.text.span.BulletSpanCompat
import net.xpece.android.text.span.TextAppearanceSpanCompat
import net.xpece.android.text.span.asSpan
import net.xpece.commons.android.sample.R
import net.xpece.commons.android.sample.databinding.ActivityMainBinding
import org.threeten.bp.LocalDateTime

class MainActivity : AppCompatActivity(), SnackbarActivity {
    override val snackbarParent: View
        get() = findViewById(android.R.id.content)
    override var snackbar: Snackbar? = null

    private var localDateTime = LocalDateTime.of(2000, 1, 31, 16, 30)!!

    override fun onCreate(savedInstanceState: Bundle?) {
        Log.d(TAG, "onCreate")
        super.onCreate(savedInstanceState)

        RippleDrawableCompatInflater.installDelegate()

        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.buttonLight.setOnClickListener {
            if (SDK_INT >= 21) {
                val context = it.context
                Intent(context, RippleActivity::class.java)
                    .let(context::startActivity)
            }
        }

        val adapter = MyPagerAdapter()
        binding.pager.adapter = adapter

        binding.pager.setEdgeEffectColorCompat(Color.RED)

        val d = dp(16)
        Log.d(TAG, "Dimension real size: " + d.toString(this))

        binding.textBullets.text = buildSpannedString {
            for (i in 0..3) {
                inSpans(
                    BulletSpanCompat(32, null, 8),
                    StrikethroughSpan(),
                    Typeface.defaultFromStyle(BOLD).asSpan()
                ) {
                    appendLine("Lorem ipsum")
                }
            }
            delete(length - 1, length)
        }
        binding.textBullets2.text = buildSpannedString {
            for (i in 0..3) {
                inSpans(
                    BulletSpanCompat(32, Color.BLACK),
                    StrikethroughSpan()
                ) {
                    appendLine("Lorem ipsum")
                }
            }
            delete(length - 1, length)
        }
        if (SDK_INT >= 28) {
            binding.textBullets3.text = buildSpannedString {
                for (i in 0..3) {
                    inSpans(
                        BulletSpan(32, Color.BLACK, 8),
                        StrikethroughSpan()
                    ) {
                        appendLine("Lorem ipsum")
                    }
                }
                delete(length - 1, length)
            }
        }

        binding.textSpanTest.text = buildSpannedString {
            append("This is ")
            inSpans(
                TextAppearanceSpanCompat(
                    binding.textSpanTest.context,
                    R.style.TextAppearance_Raleway_Thin
                )
            ) {
                append("thin ")
            }
            inSpans(
                TextAppearanceSpanCompat(
                    binding.textSpanTest.context,
                    R.style.TextAppearance_Raleway_Thin_Italic
                )
            ) {
                append("italic ")
            }
        }
    }

    internal class MyPagerAdapter : PagerAdapter() {
        override fun getCount(): Int {
            return 4
        }

        override fun isViewFromObject(view: View, `object`: Any): Boolean {
            return view === `object`
        }

        override fun instantiateItem(container: ViewGroup, position: Int): Any {
            val context = container.context
            val view: View
            if (position == 0) {
                val csl = ColorStateList.valueOf(Color.RED)
                val tp = TimePicker(context)
                tp.setSelectionDividerTintCompat(csl)
                view = tp
            } else if (position == 1) {
                val csl = ColorStateList.valueOf(Color.RED)
                val dp = DatePicker(context)
                dp.setSelectionDividerTintCompat(csl)
                view = dp
            } else {
                view = View(context)
            }
            container.addView(view)
            return view
        }

        override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
            val view = `object` as View
            container.removeView(view)
        }
    }

    companion object {
        @JvmField
        val TAG = MainActivity::class.java.simpleName
    }
}
